java-checks/src/main/java/org/sonar/java/checks/AbstractClassNoFieldShouldBeInterfaceCheck.java

```diff
@@ -74,10 +74,14 @@ private static boolean classIsAbstract(ClassTree tree) {
 
   private static boolean classHasNoFieldAndProtectedMethod(ClassTree tree) {
     for (Tree member : tree.members()) {
-      if (member.is(Tree.Kind.VARIABLE) || (member.is(Tree.Kind.METHOD) && ModifiersUtils.hasModifier(((MethodTree) member).modifiers(), Modifier.PROTECTED))) {
+      if (member.is(Tree.Kind.VARIABLE) || (member.is(Tree.Kind.METHOD) && isProtectedOrOverriding((MethodTreeImpl) member))) {
         return false;
       }
     }
     return true;
   }
+
+  private static boolean isProtectedOrOverriding(MethodTreeImpl member) {
+    return ModifiersUtils.hasModifier(member.modifiers(), Modifier.PROTECTED) || member.isOverriding();
```

Comment:
member.isOverriding can return null... therefore you have a NPE waiting for you right here.
The problem is a bit deeper than that in fact and was introduced with https://github.com/SonarSource/sonar-java/commit/7c493f55e153fc1a1f5154c8dcf8f5ff5cae9167